src/cogs/voice_recording.py

```
@@ -0,0 +1,319 @@
+    ) -> None:
+        """處理語音狀態變更事件"""
+        # 如果機器人被斷開連接，自動清理連接狀態
+        if member == self.bot.user and before.channel and not after.channel:
+            guild_id = before.channel.guild.id
+            if guild_id in self.voice_recorders:
+                recorder = self.voice_recorders[guild_id]
+                if recorder.is_connected:
+                    try:
+                        await recorder.leave_voice_channel()
+                        logger.info(f"機器人被斷開連接，自動清理連接狀態 (Guild: {guild_id})")
+                    except Exception as e:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)